assets/js/components/settings/SettingsKeyMetrics.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * SettingsKeyMetrics component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Switch } from 'googlesitekit-components';
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { Cell, Grid, Row } from '../../material-components';
+
+const { useSelect, useDispatch } = Data;
+
+export default function SettingsKeyMetrics() {
+	const keyMetricsWidgetHidden = useSelect( ( select ) =>
+		select( CORE_USER ).isKeyMetricsWidgetHidden()
+	);
+
+	const keyMetrics = useSelect( ( select ) =>
+		select( CORE_USER ).getKeyMetrics()
+	);
+
+	const { setKeyMetricsSetting, saveKeyMetricsSettings } =
+		useDispatch( CORE_USER );
+
+	const handleKeyMetricsToggle = useCallback( async () => {
+		await setKeyMetricsSetting(
+			'isWidgetHidden',
+			! keyMetricsWidgetHidden
+		);
+		await saveKeyMetricsSettings();
+	}, [
+		keyMetricsWidgetHidden,
+		saveKeyMetricsSettings,
+		setKeyMetricsSetting,
+	] );
+
+	if ( ! keyMetricsWidgetHidden === undefined || ! keyMetrics?.length ) {
+		return null;
+	}
```

Comment:
Wondering whether we should return the disabled switch instead of null to avoid the cumulative layout shift if key metrics settings are loaded with a little delay. 🤔 Wdyt?

Reply:
Thanks, @eugene-manuilov. 

I like the idea of there being a disabled switch to prevent layout shifts. However, it could be confusing to the user if there's no Key Metrics widget returned (ie. user have not chosen key metrics manually or answered the User Input questions) as in that case, the switch will remain disabled without any indication on why. 

I'd prefer to keep this as is and explore this idea in a follow up enhancement issue, where besides the disabled switch, possibly a tooltip or popover could be used to clearly convey the intent to the user. 

cc @jimmymadon

Reply:
Good point. Makes sense. Thanks.